### PR TITLE
Add full stub for Windows signals

### LINKED ISSUE #7339: Port signals to Windows
This issue is about porting `Signal` to Windows and therefore making it possible to implement other relevant features like `Process`.

Short introduction (moved from #5430):
--
[Microsft doc about signals](https://docs.microsoft.com/en-us/previous-versions/xdkz3x12(v%3Dvs.140))
I already looked at this page, and I think these signals are only there because they are part of the C standard library. From [wikipedia](https://en.wikipedia.org/wiki/C_signal_handling):

> The C standard defines only 6 signals. They are all defined in signal.h header (csignal header in C++):
> 
> SIGABRT - "abort", abnormal termination.
> SIGFPE - floating point exception.
> SIGILL - "illegal", invalid instruction.
> SIGINT - "interrupt", interactive attention request sent to the program.
> SIGSEGV - "segmentation violation", invalid memory access.
> SIGTERM - "terminate", termination request sent to the program.
> 
> Additional signals may be specified in the signal.h header by the implementation. For example, Unix and Unix-like operating systems (such as Linux) define more than 15 additional signals; see Unix signal.

And these 6 signals are the only ones "supported" by Windows.

There is already an [issue](https://github.com/nodejs/node/issues/12378) about this topic on the Node.js repo.
sam-github commented there:

> [...] Windows as an O/S has absolutely no signal support. [...]
> 
> If SIGTERM was supported on Windows, you could listen for it and receive it... but you cannot. You can listen..., but you'll never receive it.

And I guess this is exactly the case. You have a function for listening to signals (`signal()`), but there is nothing like `kill()` for sending signals on Windows.

---

So this is about finding replacement functions to "fake" signals on Windows. I already found some replacements for signals one might want to send:

- SIGTERM: `SendMessage(WM_CLOSE)` <- this is also what `taskkill.exe` uses; works for windowed processes
- SIGKILL: `TerminateProcess()` <- used by `taskkill.exe /f`; works for windowed and console processes
- SIGINT: `GenerateConsoleCtrlEvent()` <- works for console processes, simulates Ctrl + C

List of UNIX signals: https://en.wikipedia.org/wiki/Signal_(IPC)
Another microsoft doc that mentions replacements for UNIX signals: https://docs.microsoft.com/en-us/previous-versions/ms811896(v=msdn.10)#signals-and-signal-handling

### PR DESCRIPTION
**Pull request by HertzDevil**
Resolves #7339.

This defines the 6 signals required by the C standard, using their values in MSVC's headers, plus one Windows-exclusive signal, `Signal::BREAK`. The old `Signal::KILL` no longer exists, as it was only a placeholder value needed by the standard library specs to compile, and this PR removes that need. (Most POSIX signals do not exist on Windows, but technically this was already a problem for `PWR`, `STKFLT`, and `UNUSED` on non-Linux systems.)

All of `Signal`'s methods (`#trap`, `#ignore`, `#reset`) simply raise `NotImplementedError`. This is intentional, as the other platform-independent replacement APIs shall suffice for the moment.

### CODE REVIEW

File: src/signal.cr [hunk 1]
```diff
@@ -77,12 +92,17 @@ enum Signal : Int32
   # before the custom handler is called, hence a custom `CHLD` handler must
   # check child processes using `Process.exists?`. Trying to use waitpid with a
   # zero or negative value won't work.
+  #
+  # NOTE: `Process.on_interrupt` is preferred over `Signal::INT.trap`, as the
+  # former also works on Windows.
```
Comment (straight-shoota) [inline src/signal.cr:97]: ditto
```suggestion
  # NOTE: `Process.on_interrupt` is preferred over `Signal::INT.trap` as a 
  # portable alternative which also works on Windows.
```

**Review by straight-shoota — COMMENTED** [review]
Looks good. Especially moving everything into `Crystal::System`. 👍 
I only have a couple of comments on improving the wording in docs.

**Comment by HertzDevil** [comment]
Pinging @gui

**Comment by HertzDevil** [comment]
Pinging @lbguilherme to see if this move into `Crystal::System` would affect WebAssembly in any way (probably not, as the prelude still doesn't `require "signal"`)

**Comment by lbguilherme** [comment]
> Pinging @lbguilherme to see if this move into `Crystal::System` would affect WebAssembly in any way (probably not, as the things in `src/kernel.cr` are still guarded and the default prelude still doesn't `require "signal"`)

It seems fine as is, shouldn't affect anything. Actually, with your changes it could be possible to require signal on the prelude for wasm... I can test that later.

**Pull request merged**